endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
 	healthInfo := models.Health{Database: dbStatus, RabbitMQ: viper.GetString("dummy_health_rabbitmq")}
 	json.NewEncoder(w).Encode(healthInfo)
 }
+
+//Find survey by reference, short name, long name, or any combination of the three
+func getSurvey(w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    queryParams := r.URL.Query()
+    if len(queryParams) == 0 {
+        w.WriteHeader(http.StatusBadRequest)
+        errorString := models.Error{
+            Error: "No query parameters provided for search",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var sb strings.Builder
+    sb.WriteString(" WHERE 1=1")
+    for k := range queryParams {
+        switch k {
+        case "surveyRef":
+            sb.WriteString(" AND survey_ref='")
+            sb.WriteString(queryParams.Get("surveyRef"))
```

Comment:
Does this protect against sql injection? 

What happens if I add "surveyRef=1; DROP DATABASE"?

I would have expected parameterized values here like you've done with the insert?

Reply:
https://stackoverflow.com/questions/26345318/how-can-i-prevent-sql-injection-attacks-in-go-while-using-database-sql